### PR TITLE
Simplify EqualsOnSignatureLine rule

### PR DESCRIPTION
**Pull request by t-kameyama**

### CODE REVIEW

**Review by BraisGabin — APPROVED** [review]

**Review by schalkms — COMMENTED** [review]
Thanks for this PR!
I’m not sure whether those changes really simplify the rule. Could you please elaborate why? The codecov report shows an increase in complexity of a very small margin.

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/EqualsOnSignatureLine.kt [hunk 1]
```diff
@@ -37,22 +38,13 @@ class EqualsOnSignatureLine(config: Config = Config.empty) : Rule(config) {
     override val issue = Issue(javaClass.simpleName, Severity.Style, MESSAGE, Debt.FIVE_MINS)
 
     override fun visitNamedFunction(function: KtNamedFunction) {
-        val equals = function.equalsToken ?: return
-
-        // Get the tokens after the parameter list (but not including)
-        val afterParams = function.valueParameterList?.siblings(forward = true, withItself = false) ?: return
-
-        // Collect tokens until we find the equals sign
-        val untilEquals = afterParams.takeWhile { it !== equals }.toList()
```
Comment (t-kameyama) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/EqualsOnSignatureLine.kt:46]: `toList()` is useless. It should be processed as Sequence.

**Review by t-kameyama — COMMENTED** [review]
I think it's simpler to go back from the equals token.

**Comment by schalkms** [comment]
Thanks for the explanation. Let’s see what others think about the change.

**Pull request merged**